### PR TITLE
fixes strict effects

### PR DESCRIPTION
**Pull request by ringabout**

### CODE REVIEW

File: lib/system/comparisons.nim [hunk 1]
```diff
@@ -306,7 +306,7 @@ proc `==`*[T](x, y: openArray[T]): bool =
   result = true
 
 
-proc `==`*[T](x, y: seq[T]): bool {.noSideEffect.} =
+proc `==`*[T](x, y: seq[T]): bool =
```
Comment (ringabout) [inline lib/system/comparisons.nim]: The equal operator defined for children of `seqs` can have side effects.

**Comment by Araq** [comment]
This could reintroduce problems.

File: lib/system/comparisons.nim (hunk 1 above)
Comment (ringabout) [inline lib/system/comparisons.nim]:
```suggestion
proc `==`*[T](x, y: seq[T]): bool {.noSideEffect.} =
```

**Comment by ringabout** [comment]
Wait for https://github.com/nim-lang/fusion/pull/104

File: lib/system.nim [hunk 2]
```diff
@@ -1683,7 +1683,7 @@ proc pop*[T](s: var seq[T]): T {.inline, noSideEffect.} =
     result = s[L]
     setLen(s, L)
 
-func `==`*[T: tuple|object](x, y: T): bool =
+proc `==`*[T: tuple|object](x, y: T): bool =
```
Comment (ringabout) [inline lib/system.nim:1686]: It was a mistake in my previous PR, I revert it now.

**Review by Varriount — APPROVED** [review]

**Pull request merged**